### PR TITLE
Added Backend classes (7 in total)

### PR DESCRIPTION
**Pull request by hhjoel**

### CODE REVIEW

File: friendly/src/main/java/com/google/sps/Email.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    public static boolean isValidEmail(String email) {
+        return email.matches(VALIDATION_REGEX);
+    }
+
+    public String getEmail() {
+        return this.email;
+    }
+
+    @Override
+    public String toString() {
```
Comment (HeHaowei) [inline friendly/src/main/java/com/google/sps/Email.java]: Why do you need both getEmail and toString function?

**Review by HeHaowei — COMMENTED** [review]
Maybe you could put all these classes into a data folder, thus we would not mix it with the servlet, util or other classes

**Review by riku1212 — COMMENTED** [review]
LGTM, but please add the target dir in the .gitignore so it does not appear in the commit.:)

File: friendly/src/main/java/com/google/sps/Email.java (hunk 1 above)
Comment (hhjoel) [reply friendly/src/main/java/com/google/sps/Email.java]: getEmail() is the standard getter method for a private variable
toString() is overriding the toString() method in the parent class

Though they both return the same thing, it is more intuitive to call getEmail() when we want to get a private variable than toString(), so we need to getEmail() method; we need the toString() method also as it is implicitly called by some functions in the Java infrastructure

**Comment by hhjoel** [comment]
Put all files into a "model" folder as recommended by Haowei

Not sure what Inggri means?

File: friendly/src/main/java/com/google/sps/Email.java (hunk 1 above)
Comment (HeHaowei) [reply friendly/src/main/java/com/google/sps/Email.java]: Noted, thanks for explaining!

**Review by HeHaowei — APPROVED** [review]
LGTM+1!

**Comment by riku1212** [comment]
@hhjoel Nevermind, I already added the .gitignore file in my PR. What I meant is that in the current commits, the maven build `target` directory is also included, which is not useful to track.

**Pull request merged**